torchbenchmark/models/phlippe_resnet/__init__.py

```diff
@@ -0,0 +1,164 @@
+                subsample = (bc == 0 and block_idx > 0)
+                blocks.append(
+                    self.hparams.block_class(c_in=c_hidden[block_idx if not subsample else (block_idx - 1)],
+                                             act_fn=self.hparams.act_fn,
+                                             subsample=subsample,
+                                             c_out=c_hidden[block_idx])
+                )
+        self.blocks = nn.Sequential(*blocks)
+
+        # Mapping to classification output
+        self.output_net = nn.Sequential(
+            nn.AdaptiveAvgPool2d((1, 1)),
+            nn.Flatten(),
+            nn.Linear(c_hidden[-1], self.hparams.num_classes)
+        )
+
+    def _init_params(self):
+        # Based on our discussion in Tutorial 4, we should initialize the convolutions according to the activation function
+        # Fan-out focuses on the gradient distribution, and is commonly used in ResNets
+        for m in self.modules():
+            if isinstance(m, nn.Conv2d):
+                nn.init.kaiming_normal_(
+                    m.weight, mode='fan_out', nonlinearity=self.hparams.act_fn_name)
+            elif isinstance(m, nn.BatchNorm2d):
+                nn.init.constant_(m.weight, 1)
+                nn.init.constant_(m.bias, 0)
+
+    def forward(self, x):
+        x = self.input_net(x)
+        x = self.blocks(x)
+        x = self.output_net(x)
+        return x
+
+
+class Model(BenchmarkModel):
+    task = COMPUTER_VISION.CLASSIFICATION
+    DEFAULT_TRAIN_BSIZE = 128
+    DEFAULT_EVAL_BSIZE = 128
+
+    def __init__(self, test, device, jit=False, batch_size=DEFAULT_TRAIN_BSIZE, extra_args=[]):
+        super().__init__(test=test, device=device, jit=jit,
+                         batch_size=batch_size, extra_args=extra_args)
+        self.model = ResNetModel()
+        self.model.to(device)
+        self.example_inputs = (
+            torch.randn((self.batch_size, 3, 32, 32), device=self.device),
+        )
+        self.example_target = torch.randint(0, 10, (self.batch_size,), device=self.device)
+        dataset = data.TensorDataset(self.example_inputs[0], self.example_target)
+        dummy_loader = data.DataLoader(dataset, batch_size=self.batch_size)
+        self.optimizer = optim.SGD(self.model.parameters(), lr=0.1, momentum=0.9, weight_decay=1e-4)
+        self.criterion = nn.CrossEntropyLoss()
+
+    def get_module(self):
+        return self.model, self.example_inputs
+
+    def train(self):
+        model = self.model
+        (images, ) = self.example_inputs
+        model.train()
+        targets = self.example_target
+        output = model(images)
+        loss = self.criterion(output, targets)
+        loss.backward()
+        self.optimizer.step()
+        self.optimizer.zero_grad()
+    
+    def eval(self):
+        model = self.model
+        (images, ) = self.example_inputs
+        model.eval()
```

Comment:
Let's move this to init

Reply:
fixed by 70289f205a021189a88f830b7de49d48a44b306e